### PR TITLE
feat: required changes relate to HTTP resource

### PR DESCRIPTION
**Pull request by ashraf706**
fixes: https://github.com/gravitee-io/issues/issues/7517
Add template engine in factor context
Add factor context arguments as required
Implement error handler and template engine

### CODE REVIEW

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html [hunk 1]
```diff
@@ -55,11 +55,25 @@ <h5>Configuration</h5>
...
         <p>
           Resource is a configuration facility to reuse some settings.
         </p>
       </div>
+
+      <div class="gv-page-description-content" *ngIf="isHttpResource()">
+        <h4>HTTP Resource</h4>
+        <p>
+          This resource facilitates HTTP requests to send and verify verification code.
+          It supports AM Expression Language (EL) to configure the resource properties.
```
Comment (remisultan) [inline gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html:68]:
```suggestion
          It supports Gravitee Expression Language (EL) to configure the resource properties.
```

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.ts [hunk 2]
```diff
@@ -88,4 +88,13 @@ export class ResourceComponent implements OnInit {
...
   }
+
+  isHttpResource() {
+    let retVal = false;
+    if (this.resourceSchema) {
+      const resourceId = this.resourceSchema.id;
+      retVal = resourceId ? resourceId.includes('HttpResourceConfiguration') : false;
+    }
+    return retVal;
+  }
```
Comment (remisultan) [inline gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.ts:99]:
```suggestion
  isHttpResource() {
    if (this.resourceSchema) {
      const resourceId = this.resourceSchema.id;
      return resourceId ? resourceId.includes('HttpResourceConfiguration') : false;
    }
    return false;
  }
```

**Review by remisultan — CHANGES_REQUESTED** [review]

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html [hunk 3]
```diff
@@ -55,11 +55,25 @@ <h5>Configuration</h5>
...
+      <div class="gv-page-description-content" *ngIf="!isHttpResource()">
         <p>
           Resource is a configuration facility to reuse some settings.
         </p>
       </div>
+
+      <div class="gv-page-description-content" *ngIf="isHttpResource()">
+        <h4>HTTP Resource</h4>
+        <p>
+          This resource facilitates HTTP requests to send and verify verification code.
```
Comment (farmborough) [inline gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html:67]:
```suggestion
          This resource facilitates HTTP requests to send and verify a generated, one time code.
```

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html [hunk 4]
```diff
@@ -55,11 +55,25 @@ <h5>Configuration</h5>
...
+          It supports AM Expression Language (EL) to configure the resource properties.
+        </p>
+        <h5>HTTP Response</h5>
+        <p>Response is accessible with EL such as: <br><b>&#123;'#xpath(#response.content, ".//code") == '1234'&#125;'</b><br> where code is a property of the XML response.</p>
```
Comment (farmborough) [inline gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html:71]:
```suggestion
        <p>The response can be accessed via EL using xpath, e.g.: <br><b>&#123;'#xpath(#response.content, ".//code") == '1234'&#125;'</b><br> where code is a property of the XML response.</p>
```

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html [hunk 5]
```diff
@@ -55,11 +55,25 @@ <h5>Configuration</h5>
...
+        <h5>Pin or Verification Code</h5>
+        <p>
+          User pin or verification code is available in <b>' &#123;#verificationCode &#125;'</b> EL property once user enters the code during the verify step.
```
Comment (farmborough) [inline gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html:74]:
```suggestion
          The user pin or verification code is available in the<b>' &#123;#verificationCode &#125;'</b> EL property after the user has submitted the code for verification.
```

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/RootProvider.java [hunk 6]
```diff
@@ -344,6 +345,8 @@ protected void doStart() throws Exception {
                 .handler(clientRequestParseHandler)
                 .handler(rememberDeviceSettingsHandler)
                 .handler(new MFAChallengeEndpoint(factorManager, userService, thymeleafTemplateEngine, deviceService, applicationContext, domain));
+        rootRouter.route(PATH_MFA_CHALLENGE)
+                .failureHandler(new MFAChallengeFailureHandler(authenticationFlowContextService));
```
[the remark this answers is not in the document]
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/RootProvider.java]: done

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html (hunk 1 above)
Comment (ashraf706) [reply gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html]: done

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.ts (hunk 2 above)
Comment (ashraf706) [reply gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.ts]: done

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html (hunk 4 above)
Comment (ashraf706) [reply gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html]: done

File: gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html (hunk 5 above)
Comment (ashraf706) [reply gravitee-am-ui/src/app/domain/settings/resources/resource/resource.component.html]: done

**Review by remisultan — APPROVED** [review]
Usage is fine LGTM

I think the little note on the side regarding the `xpath` expression can be misleading.
There are single quotes here and there that do not make sense

image

File: pom.xml [hunk 7]
```diff
@@ -118,7 +118,7 @@
         <gravitee-policy-ipfiltering.version>1.8.0</gravitee-policy-ipfiltering.version>
         <gravitee-policy-request-validation.version>1.12.0</gravitee-policy-request-validation.version>
         <gravitee-policy-latency.version>1.4.0</gravitee-policy-latency.version>
-        <gravitee-license-node.version>1.2.1</gravitee-license-node.version>
+        <gravitee-license-node.version>1.1.0</gravitee-license-node.version>
```
Comment (remisultan) [inline pom.xml:121]: Why the license change here ?

**Review by remisultan — APPROVED** [review]

File: pom.xml (hunk 7 above)
Comment (remisultan) [reply pom.xml:121]: seen with @ashraf706  already

**Pull request merged**